### PR TITLE
Cleanup Cython structs

### PR DESCRIPTION
**Pull request by svlandeg**
Following up on https://github.com/explosion/spaCy/pull/9150

## Description
Removing some attributes that we were keeping around but are no longer necessary.

### Types of change
cleanup

## Checklist
- [x] I confirm that I have the right to submit this contribution under the project's MIT license.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

**Comment by adrianeboyd** [comment]
I was just looking a bit more at the symbols, and I think that renumbering the symbols has the potential to break saved `.spacy` files, so I think we're kind of stuck with the current symbol numbering. We can rename `IS_OOV_DEPRECATED` to `FLAG##`?

**Comment by svlandeg** [comment]
You're right that it makes no sense to break existing `.spacy` files if we can help it - I'll update.

**Comment by adrianeboyd** [comment]
Actually, we can't change any of the symbols without potentially breaking things, ugh. I think I'll try to add them to the next version of `.spacy` so there would be more freedom to modify them in the future without breaking saved files.

File: spacy/tokenizer.pxd [hunk 1]
```diff
@@ -23,11 +24,7 @@ cdef class Tokenizer:
     cdef object _infix_finditer
     cdef object _rules
     cdef PhraseMatcher _special_matcher
-    # TODO convert to bool in v4
-    cdef int _faster_heuristics
-    # TODO next one is unused and should be removed in v4
-    # https://github.com/explosion/spaCy/pull/9150
-    cdef int _unused_int2
+    cdef bool _faster_heuristics
```
Comment (adrianeboyd) [inline spacy/tokenizer.pxd:27]:
```suggestion
    cdef bint _faster_heuristics
```

(and no import)

**Pull request merged**